jnigen/test/bindings_test.dart

```diff
@@ -149,4 +155,126 @@ void main() async {
   test('exceptions', () {
     expect(() => Example.throwException(), throwsException);
   });
+  group('generics', () {
+    test('MyStack<T>', () {
+      using((arena) {
+        final stack = MyStack(JString.type)..deletedIn(arena);
+        stack.push('Hello'.toJString()..deletedIn(arena));
+        stack.push('World'.toJString()..deletedIn(arena));
+        expect(stack.pop().toDartString(deleteOriginal: true), 'World');
+        expect(stack.pop().toDartString(deleteOriginal: true), 'Hello');
+      });
+    });
+    test('MyMap<K, V>', () {
+      using((arena) {
+        final map = MyMap(JString.type, Example.type)..deletedIn(arena);
+        final helloExample = Example()
```

Comment:
Tiny nit: this example becomes slightly more readable if the field is set in the constructor.